src/core.js

```diff
@@ -38,56 +38,61 @@ QUnit.isLocal = ( window && window.location && window.location.protocol === "fil
 QUnit.version = "@VERSION";
 
 extend( QUnit, {
+	config,
+
+	dump,
+	equiv,
+	is,
+	objectType,
 	on,
+	onError,
+	onUnhandledRejection,
+	pushFailure,
 
+	assert: Assert.prototype,
 	module,
-
-	test: test,
+	test,
 
 	// alias other test flavors for easy access
 	todo: test.todo,
 	skip: test.skip,
 	only: test.only,
 
 	start: function( count ) {
-		var globalStartAlreadyCalled = globalStartCalled;
 
-		if ( !config.current ) {
-			globalStartCalled = true;
-
-			if ( runStarted ) {
-				throw new Error( "Called start() while test already started running" );
-			} else if ( globalStartAlreadyCalled || count > 1 ) {
-				throw new Error( "Called start() outside of a test context too many times" );
-			} else if ( config.autostart ) {
-				throw new Error( "Called start() outside of a test context when " +
-					"QUnit.config.autostart was true" );
-			} else if ( !config.pageLoaded ) {
-
-				// The page isn't completely loaded yet, so we set autostart and then
-				// load if we're in Node or wait for the browser's load event.
-				config.autostart = true;
-
-				// Starts from Node even if .load was not previously called. We still return
-				// early otherwise we'll wind up "beginning" twice.
-				if ( !document ) {
-					QUnit.load();
-				}
-
-				return;
-			}
-		} else {
+		if ( config.current ) {
 			throw new Error( "QUnit.start cannot be called inside a test context." );
 		}
 
-		scheduleBegin();
-	},
+		var globalStartAlreadyCalled = globalStartCalled;
+		globalStartCalled = true;
 
-	config: config,
+		if ( runStarted ) {
+			throw new Error( "Called start() while test already started running" );
+		}
+		if ( globalStartAlreadyCalled || count > 1 ) {
+			throw new Error( "Called start() outside of a test context too many times" );
+		}
+		if ( config.autostart ) {
+			throw new Error( "Called start() outside of a test context when " +
+				"QUnit.config.autostart was true" );
+		}
 
-	is: is,
+		if ( config.pageLoaded ) {
+			scheduleBegin();
+			return;
```

Comment:
I'm a fan of early returns for handling errors and unusual scenarios, but I think this might have taken it a bit too far as it is it now losing structure of the logic as well. Browser and Node js handling are both the "main" purpose of this function and I would expect those to be part of the same non-early part of the function body.

When extending the function it should be easy to add something to the end that both branches share, which would be difficult in this case as its natural end is now copied in two places. I don't really mind how much "work" it would be to add something to this function, that's easy to refactor if needed. The concern is that when reading this code now, it doesnt' resemble the logical decision tree that the function is responsible for, I think?

Anyway, my suggestion would be to omit this last one and use an else branch. Possibly reversing the order so that the non-ready case remains handled "first" like before.